lib/Sema/DerivedConformances.cpp

```diff
@@ -35,12 +35,16 @@ DeclContext *DerivedConformance::getConformanceContext() const {
   return cast<DeclContext>(ConformanceDecl);
 }
 
-void DerivedConformance::addMembersToConformanceContext(
+bool DerivedConformance::addMembersToConformanceContext(
     ArrayRef<Decl *> children) {
   auto IDC = cast<IterableDeclContext>(ConformanceDecl);
+  bool anyInvalid = false;
   for (auto child : children) {
     IDC->addMember(child);
+    TypeChecker::typeCheckDecl(child);
```

Comment:
This is going to do extra work in secondary files and slow down compilation. Instead, you should check conformances before visiting members in the DeclChecker, then the ordinary member walk in `visit(Class|Struct|Enum)Decl()` will handle them.